src/init/features/functions/index.ts

```diff
@@ -30,6 +26,106 @@ export async function doSetup(setup: any, config: any, options: Options) {
       ensure(projectId, "runtimeconfig.googleapis.com", "unused", true),
     ]);
   }
+  setup.functions = {};
+  if (!config.src.functions) {
+    // if functions have not been initialized yet
```

Comment:
I will consider removing this comment if you think it falls under [go/tott/563](https://goto.google.com/tott/563).

Reply:
not sure actually. I don't know if it's completely obvious that checking `(!config.src.functions)` means I'm checking if cloud functions has been initialized yet. It also feels unnecessary to refactor this out into a function  `isFunctionsInitialized` (or something like that) which will end up being 1 line of code. Ideally, there would be some state in `Config` that tracks whether `functions` has been initialized which I could check here. Maybe I will leave it for now, and revisit this question of "initialized" state in the future?

Reply:
Those are some very helpful tips about writing comments, btw!